src/components/UserProfile/UserSettings/UserGeneralSettings/index.tsx

```diff
@@ -57,6 +57,11 @@ const UserGeneralSettings: React.FC = () => {
     user ? `/api/v1/user/${user?.id}/settings/main` : null
   );
 
+  useEffect(() => {
+    setMovieQuotaEnabled(!!data?.movieQuotaLimit && !!data?.movieQuotaDays);
+    setTvQuotaEnabled(!!data?.tvQuotaLimit && !!data?.tvQuotaDays);
```

Comment:
You must check if the limits are null/undefined here or unlimited won't check the box.